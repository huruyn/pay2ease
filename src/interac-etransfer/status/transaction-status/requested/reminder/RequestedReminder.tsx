import
{
  Button,
  Col,
  Row,
  Stack,
} from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import NumberFormat from 'react-number-format';
import { TransactionProps } from '../../TransactionStatus';

function RequestedReminder({
  transaction,
}: TransactionProps) {
  const navigate = useNavigate();

  return (
    <>
      <Row>
        <Col>
          <ul>
            <li>
              The request for money was sent but the contact has not yet sent the transfer.
            </li>
            <li>
              You can send your contact a reminder or cancel the request.
            </li>
          </ul>
        </Col>
      </Row>
      <Row>
        <Col xs={4}>To</Col>
        <Col xs={6}>
          <strong>{transaction ? `${transaction.contact.firstName || ''} ${transaction.contact.lastName || ''}` : ''}</strong>
          <div>{transaction && ` (${transaction.contact.email})`}</div>
        </Col>
      </Row>
      <Row>
        <Col xs={4}>Transfer Amount</Col>
        <Col xs={6}>
          <strong className="amount">
            {transaction && transaction.amount
            && (
              <NumberFormat
                value={transaction.amount}
                defaultValue={0}
                displayType="text"
                prefix="$ "
                suffix=" CAD"
                thousandSeparator
                decimalScale={2}
                fixedDecimalScale
              />
            )}
          </strong>
        </Col>
      </Row>
      <Stack gap={3} direction="horizontal">
        <Button
          className="zippy-btn zippy-flat d-flex"
          onClick={() => navigate('/interac-etransfer/status/requested')}
        >
          Back to Status
        </Button>
        <Button
          className="zippy-btn d-flex ms-auto"
          onClick={() => navigate('/interac-etransfer/request-money')}
        >
          Send Another Request
        </Button>
      </Stack>
    </>
  );
}

export default RequestedReminder;
