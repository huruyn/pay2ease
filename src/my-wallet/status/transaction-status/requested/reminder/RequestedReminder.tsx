import
{
  Button,
  Col,
  Row,
} from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import NumberFormat from 'react-number-format';
import { TransactionProps } from '../../TransactionStatus';

function RequestedReminder({
  transaction,
}: TransactionProps) {
  const navigate = useNavigate();

  return (
    <>
      <Row>
        <Col xs={4}>To</Col>
        <Col xs={8}>
          <strong>
            {transaction
              ? `${transaction.contact.firstName || ''} ${
                transaction.contact.lastName || ''
              }`
              : ''}
          </strong>
          <div>{transaction && ` (${transaction.contact.email})`}</div>
        </Col>
      </Row>
      <Row>
        <Col xs={4}>Transfer Amount</Col>
        <Col xs={8}>
          <strong className="amount">
            {transaction && transaction.amount && (
              <NumberFormat
                value={transaction.amount}
                defaultValue={0}
                displayType="text"
                prefix="$ "
                suffix=" CAD"
                thousandSeparator
                decimalScale={2}
                fixedDecimalScale
              />
            )}
          </strong>
        </Col>
      </Row>
      <Row className="transfer-btn">
        <Button
          className="zippy-btn d-flex ms-auto"
          onClick={() => navigate('/my-wallet/zippy-money/request/transaction-start')}
        >
          Send Another Request
        </Button>
      </Row>
    </>
  );
}

export default RequestedReminder;
