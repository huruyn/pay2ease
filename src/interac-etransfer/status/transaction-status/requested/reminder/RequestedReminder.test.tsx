import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RequestedReminder from './RequestedReminder';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('RequestedReminder Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <RequestedReminder
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render RequestedReminder', () => {
    render(
      <BrowserRouter>
        <RequestedReminder
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
        />
      </BrowserRouter>,
    );

    expect(screen.getByText('To')).toBeInTheDocument();
    expect(screen.getByText('Transfer Amount')).toBeInTheDocument();
  });
});
