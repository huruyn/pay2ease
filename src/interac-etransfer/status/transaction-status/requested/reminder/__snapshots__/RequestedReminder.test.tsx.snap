// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestedReminder Component matches the snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col"
    >
      <ul>
        <li>
          The request for money was sent but the contact has not yet sent the transfer.
        </li>
        <li>
          You can send your contact a reminder or cancel the request.
        </li>
      </ul>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      To
    </div>
    <div
      className="col-6"
    >
      <strong>
        
      </strong>
      <div />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      Transfer Amount
    </div>
    <div
      className="col-6"
    >
      <strong
        className="amount"
      />
    </div>
  </div>,
  <div
    className="hstack gap-3"
  >
    <button
      className="zippy-btn zippy-flat d-flex btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Back to Status
    </button>
    <button
      className="zippy-btn d-flex ms-auto btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Send Another Request
    </button>
  </div>,
]
`;
